### PR TITLE
deprecate checkRequiredPermission()

### PR DESCRIPTION
**Pull request by marcardar**

### CODE REVIEW

**Comment by marcardar** [comment]
Updated checkRequiredPermission() to be based on the SDK level.

**Comment by timrae** [comment]
Thanks, it looks fine, please rebase onto develop and squash.

File: api/src/main/java/com/ichi2/anki/api/AddContentApi.java [hunk 1]
```diff
@@ -623,15 +629,11 @@ public static String getAnkiDroidPackageName(Context context) {
      * Get the ID for any permission which is required to use the API
      * @param context a Context that can be used to get the PackageManager
      * @return id of a permission required to access the API or null if no permission is required
+     * @deprecated use {@link #READ_WRITE_PERMISSION} instead and see class-level docs
      */
+    @Deprecated
     public static String checkRequiredPermission(Context context) {
```
Comment (marcardar) [inline api/src/main/java/com/ichi2/anki/api/AddContentApi.java]: Maybe just remove this?

Comment (timrae) [inline api/src/main/java/com/ichi2/anki/api/AddContentApi.java]: Remove the method entirely? Nah let's deprecate it for now and remove later.

**Comment by marcardar** [comment]
Done.

File: api/src/main/java/com/ichi2/anki/api/AddContentApi.java (hunk 1 above)
Comment (marcardar) [inline api/src/main/java/com/ichi2/anki/api/AddContentApi.java]: Okay. BTW I wouldn't be too worried about making breaking changes like this. Clients are not automatically updated to the latest API. They have to increment the version in gradle, and then they can decide whether it's worth making the small changes or sticking with their existing version.  This minor pain should be weighed against new clients seeing a simpler API.

Comment (timrae) [inline api/src/main/java/com/ichi2/anki/api/AddContentApi.java]: Yeah I guess so, but so far we've been deprecating the methods. Maybe before releasing v2 I'll pull out all the deprecated methods and compile a changelog.

**Pull request merged**